libs/state/src/lib/convert-partials-to.observable.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  combineLatest,
+  isObservable,
+  Observable,
+  startWith,
+  timer,
+} from 'rxjs';
+import { Injector, isSignal, Signal } from '@angular/core';
+import { toObservable } from '@angular/core/rxjs-interop';
+import { map } from 'rxjs/operators';
+import { coalesceWith } from '@rx-angular/cdk/coalescing';
+
+export function convertPartialsToObservable<
+  T extends object,
+  K extends keyof T
+>(
+  input: Partial<{ [K: string]: Observable<T[K]> | Signal<T[K]> }>,
+  injector?: Injector
+): Observable<Partial<T>> {
+  const keys = Object.keys(input);
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+  const observables = keys.map((key) => {
+    const obsOrSignal = input[key];
+
+    if (isObservable(obsOrSignal)) {
+      return obsOrSignal?.pipe(startWith(undefined));
+    }
+    if (isSignal(obsOrSignal)) {
+      return toObservable(obsOrSignal, { injector });
+    }
+  });
+
+  return combineLatest(observables).pipe(
+    map((values) => {
+      const result: Partial<T> = {} as Partial<T>;
+      for (let i = 0; i < keys.length; i++) {
+        // @ts-ignore
+        result[keys[i]] = values[i];
+      }
+      return result;
+    }),
+    coalesceWith(timer(0))
```

Comment:
is this the correct coalescing operator?

Reply:
it is the correct operator, but why would you coalesce it on the next `macrotask`? It makes much more sense to coalesce on `microtask` instead. Also, it shouldn't trigger ngZone.

Please try with `coalesceWith(from(Promise.resolve()))` and use the unpatched promise from the rx angular cdk.

We have to write tests though, that confirm `RxState` is not running out of zone. That could potentially cause issues when state selections are running out of zone.

Reply:
also, I think coalescing should happen before the mapping. This way, the mapping would be less often executed

Reply:
makes absolutely sense, will rework it soon